bin/render-activity-stream-html.js

```diff
@@ -1,18 +1,176 @@
- /* eslint-disable no-sync */
+ /* eslint-disable no-console, no-sync */
 const fs = require("fs");
+const {mkdir} = require("shelljs");
 const path = require("path");
 
-const BASE_FILE_PATH = path.resolve(__dirname, "../system-addon/data/content/");
-
 // Note: this file is generated by webpack from system-addon/content-src/activity-stream-prerender.jsx
 const prerender = require("./prerender");
 
+const DEFAULT_LOCALE = "en-US";
 const DEFAULT_OPTIONS = {
-  baseUrl: "resource://activity-stream/",
-  locale: "", // TODO: pass in options.locale. For now, we're just doing empty strings.
-  title: ""
+  addonPath: "../system-addon",
+  baseUrl: "resource://activity-stream/"
 };
 
+// From https://hg.mozilla.org/l10n-central/
+// console.log(`const MOZILLA_CENTRAL_LOCALES = ${JSON.stringify([...document.querySelectorAll("a.list:not([href*='x-testing']")].map(a => a.textContent.trim()), null, 2)};`)
+const MOZILLA_CENTRAL_LOCALES = [
```

Comment:
So… "ia" was added since I originally created a local branch 1 month ago, and it has 100% strings localized for activity stream: https://pontoon.mozilla.org/ia/activity-stream-new-tab/strings.properties/

Would be unfortunate if we didn't notice and missed the prerendering for that locale. But on that note, if we have strings for a locale that we aren't building, we should be able to notice right away !

Should it break this render script? Big warning? I suppose the strings import script could be smarter, but then we'll want to share this array outside of this file.

Or alternatively, we can dynamically augment MOZILLA_CENTRAL_LOCALES with the locales that we have localized strings.

Actually, I think I'll go with the dynamic augment and big warning.

And just to be explicit, we can't rely on our own locale list from locales.json as we would have missed out on bn-IN, ja-JP-mac, ta-LK which we have similar strings to reuse from bn-BD, ja, ta respectively.

Reply:
Added a comment documenting the purpose of this list.